kit/ForKit.cpp

```diff
@@ -383,18 +380,28 @@ static int createLibreOfficeKit(const std::string& childRoot,
     LOG_DBG("Forking a coolkit process with jailId: " << jailId << " as spare coolkit #"
                                                       << spareKitId << '.');
     const auto startForkingTime = std::chrono::steady_clock::now();
-
-    const pid_t pid = fork();
-    if (!pid)
+    pid_t pid = 0;
+    if (Util::isKitInProcess() && !lokitProcessIsRunning)
     {
-        // Child
-
-        // Close the pipe from coolwsd
-        close(0);
+        std::thread ([childRoot, jailId, sysTemplate, loTemplate, queryVersion]
+            {
+                Util::setThreadName("kit_in_process");
+                lokit_main(childRoot, jailId, sysTemplate, loTemplate, NoCapsForKit, NoSeccomp,
+                   queryVersion, DisplayVersion, spareKitId);
+            }).detach();
+        lokitProcessIsRunning = true;
+    }
+    else {
+        pid = fork();
```

Comment:
Please run clang-format, or apply proper formatting. This is all wrong.

Reply:
thanks @Ashod i updated the formatting mistakes but there might be some still present sorry for that. i am still working on this patch so will update all formatting mistakes  soon...